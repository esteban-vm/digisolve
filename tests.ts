import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
