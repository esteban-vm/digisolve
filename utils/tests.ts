import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
