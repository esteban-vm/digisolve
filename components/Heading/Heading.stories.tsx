import type { Meta, StoryObj } from '@storybook/react'
import { styled } from '@/utils/styles'
import Heading from './Heading'

const Wrapper = styled.div`
  text-align: center;
  background-color: var(--color-light);
`

export default {
  component: Heading,
  title: 'Components/Heading',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
} as Meta<typeof Heading>

export const HeadingExample: StoryObj<typeof Heading> = { args: { heading: 'Test' } }
