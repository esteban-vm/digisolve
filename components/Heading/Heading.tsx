import type { FunctionalComponent } from '@/types'
import { styled } from '@/utils/styles'

/** Properties for heading component */
type HeadingProps = {
  /** The heading text */
  heading: string
  /** The subheading text */
  subheading?: string
}

const HeadingComponent: FunctionalComponent<HeadingProps> = ({ heading, subheading, ...rest }) => {
  return (
    <div {...rest}>
      <h2>{heading}</h2>
      <h3>{subheading}</h3>
    </div>
  )
}

const Heading = styled(HeadingComponent)`
  h2 {
    font-size: 250%;
    color: var(--color-black);
    margin-bottom: 20px;
    margin-top: 150px;
  }

  h3 {
    font-size: 150%;
    color: var(--color-grey);
    font-weight: 300;
    margin-bottom: 30px;
    line-height: 1.5;

    ::after {
      content: ' ';
      display: block;
      height: 2px;
      width: 100px;
      background-color: var(--color-primary);
      margin: 30px auto 0;
    }
  }
`

Heading.defaultProps = {
  className: 'row',
  subheading: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.',
}

export default Heading
