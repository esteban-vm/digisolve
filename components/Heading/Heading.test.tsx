import type { HeadingComponentProps } from '@/types'
import { render, cleanup, screen, create } from '@/tests'
import Heading from './Heading'

describe('ðŸ§ª HEADING:', () => {
  const testHeading: HeadingComponentProps = { heading: 'Test heading' }

  describe('display tests:', () => {
    let parent: HTMLElement

    beforeEach(() => {
      render(<Heading {...testHeading} />)
      parent = screen.getByRole('none')
    })

    afterEach(cleanup)

    it('should display the heading text', () => {
      const heading = screen.getByRole('heading', { name: testHeading.heading, level: 2 })
      expect(heading).toBeInTheDocument()
      expect(heading).toBeVisible()
      expect(parent).toContainElement(heading)
    })

    it('should display the subheading text', () => {
      const subheading = screen.getByRole('heading', { name: /^lorem ipsum/i, level: 3 })
      expect(subheading).toBeInTheDocument()
      expect(subheading).toBeVisible()
      expect(parent).toContainElement(subheading)
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<Heading {...testHeading} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
