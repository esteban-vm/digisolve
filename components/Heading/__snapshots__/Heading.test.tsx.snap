// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª HEADING: style tests: should render with correct styles 1`] = `
.emotion-0 h2 {
  font-size: 250%;
  color: var(--color-black);
  margin-bottom: 20px;
  margin-top: 150px;
}

.emotion-0 h3 {
  font-size: 150%;
  color: var(--color-grey);
  font-weight: 300;
  margin-bottom: 30px;
  line-height: 1.5;
}

.emotion-0 h3::after {
  content: ' ';
  display: block;
  height: 2px;
  width: 100px;
  background-color: var(--color-primary);
  margin: 30px auto 0;
}

<div
  className="row emotion-0"
  role="none"
>
  <h2>
    Test heading
  </h2>
  <h3>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
  </h3>
</div>
`;
